rust/xaynet-ffi/src/lib.rs

```diff
@@ -1,812 +1,710 @@
-//! A C-API to communicate model updates between a XayNet participant and an application.
-//!
-//! # Workflow
-//! 1. Initialize a [`Client`] with [`new_client()`]. The [`Client`] takes care of the
-//!    [`Participant`]'s PET protocol work as well as the networking with the [`Coordinator`].
-//! 2. Start the execution of the [`Client`]'s tasks with [`run_client()`].
-//! 3. Optionally request status information:
-//!     - [`is_next_round()`] indicates if another round of the PET protocol has started.
-//!     - [`has_next_model()`] indicates if another global model is available.
-//!     - [`is_update_participant()`] indicates if this [`Participant`] is eligible to submit a
-//!       trained local model in the current round.
-//! 4. Create a new zero-initialized model with [`new_model()`] or get the latest global model with
-//!    [`get_model()`]. Currently, the primitive data types [`f32`], [`f64`], [`i32`] and [`i64`]
-//!    are supported. The functions return a fat pointer [`PrimitiveModel`] to the cached primitive
-//!    model, whereas the primitive model itself is cached within the [`Client`]. The cached
-//!    primitive model can then be modified in place, for example for training. The slice is valid
-//!    across the FFI-boundary until one of the following happens:
-//!    - [`new_model()`] reallocates the memory to which [`PrimitiveModel`] points to.
-//!    - [`get_model()`] reallocates the memory to which [`PrimitiveModel`] points to if a new
-//!      global model is available since the last call to [`get_model()`].
-//!    - [`update_model()`] frees the memory to which [`PrimitiveModel`] points to.
-//!    - [`drop_model()`] frees the memory to which [`PrimitiveModel`] points to.
-//!    - [`drop_client()`] frees the memory of the [`Client`] including the model.
-//! 5. Register the cached model as an updated local model with [`update_model()`].
-//! 6. Stop and destroy the [`Client`] with [`drop_client()`].
+//! A C-API of the Xaynet mobile client.
 //!
 //! # Safety
+//!
 //! Many functions of this module are marked as `unsafe` to explicitly announce the possible
 //! unsafety of the function body as well as the return value to the caller. At the same time,
 //! each `unsafe fn` uses `unsafe` blocks to precisely pinpoint the sources of unsafety for
 //! reviewers (redundancy warnings will be fixed by [#69173]).
 //!
-//! **Note, that the `unsafe` code has not been externally audited yet!**
+//! Most of the functions have one / or more pointers as parameters.
+//! All functions ensure null-safety for these pointers.
```

Comment:
Does "null-safety" refer to a well defined Rust or `ffi-support` concept, or does it just mean that we check whether the pointer is null and return an error if it is?